scripts/web/blueprints/grading.py

```
@@ -0,0 +1,692 @@
+                        mp      = float(item.get("max_points", 0) or 0)
+                        if not awarded and pct and mp:
+                            awarded = _snap_to_grade_band(pct, mp)
+                        score += awarded
+                else:
+                    score = float(result.get("overall_score", 0) or 0)
+                feedback = str(result.get("overall_feedback", "") or "").strip()
+            score = round(min(total_max, max(0.0, score)), 1)
+            ws.cell(row, score_col).value    = score
+            ws.cell(row, feedback_col).value = feedback
+            graded += 1
+        except Exception as exc:
```

Ruff (0.15.11)

[warning] Do not catch blind exception: `Exception`

(BLE001)